packages/react-ape/renderer/elements/Button.js

```diff
@@ -0,0 +1,189 @@
+    */
+  accessibilityLabel?: ?string,
+
+  /**
+    * If true, disable all interactions for this component.
+    */
+  disabled?: ?boolean,
+
+  /**
+    * Used to locate this view in end-to-end tests.
+    */
+  testID?: ?string,
+|};
+
+function renderButton(props: ButtonProps, apeContext, parentLayout) {
+  //
+  console.log('[RENDER]');
+  const {spatialGeometry = {x: 0, y: 0}} = parentLayout;
+  const {ctx} = apeContext;
+
+  // If is relative and x and y haven't be processed, don't render
+  if (!spatialGeometry) return null;
+  // start drawing the canvas
+  const {title, color} = props;
+  const borderRadius = ButtonDefaults.containerStyle.borderRadius;
+  const backgroundColor = ButtonDefaults.containerStyle.backgroundColor;
+  let x = spatialGeometry.x || 0;
+  let y = spatialGeometry.y || 0;
+  let width = x + y;
+  let height = ButtonDefaults.containerStyle.height;
+  let globalStyle = {
+    width: width,
+    height: height,
+    color: color,
+    borderRadius: borderRadius,
+    backgroundColor,
+    lineWidth: 0,
+    borderColor: 'transparent',
+  };
+  const resetStyle = newStyle => {
+    globalStyle = {...globalStyle, newStyle};
+    console.log('style)))))', globalStyle);
+  };
+  const redrawButton = ctx => {
+    // TODO reset Style on focus
+    let newStyle = {
+      lineWidth: 2,
+      borderColor: '#ccc',
+    };
+    //let prevStyle = globalStyle
+    resetStyle(newStyle);
+  };
+
+  ctx.beginPath();
+  ctx.fillStyle = backgroundColor;
+  ctx.moveTo(x, y);
+  /**
+*  Top Right Radius
+*/
+  ctx.lineTo(x + width - borderRadius, y);
+  ctx.quadraticCurveTo(x + width, y, x + width, y + borderRadius);
+  /**
+*  Bottom right Radius
+*/
+
+  ctx.lineTo(x + width, y + height - borderRadius);
+  ctx.quadraticCurveTo(
+    x + width,
+    y + height,
+    x + width - borderRadius,
+    y + height
+  );
+
+  /**
+*  Bottom Left Radius
+*/
+  ctx.lineTo(x + borderRadius, y + height);
+  ctx.quadraticCurveTo(x, y + height, x, y + height - borderRadius);
+  /** Top left Radius */
+  ctx.lineTo(x, y + borderRadius);
+  ctx.quadraticCurveTo(x, y, x + borderRadius, y);
+
+  ctx.fill();
+  ctx.lineWidth = globalStyle.lineWidth;
+  ctx.strokeStyle = globalStyle.borderColor;
+  ctx.stroke();
+  ctx.fillStyle = color || ButtonDefaults.textStyle.color;
+  ctx.font = `${ButtonDefaults.textStyle.fontSize} Helvetica`;
+  //ctx.fillText('Start', x+ width/2 , y + height / 2);
+  ctx.textAlign = 'center';
+  ctx.fillText(title, x + width / 2, y + height / 2);
+  ctx.closePath();
+
+  const onClick = (event: SyntheticMouseEvent<HTMLButtonElement>) => {
+    const rect = {
+      x,
+      y,
+      height,
+      width,
+    };
+    const mousePosition = trackMousePosition(ctx.canvas, event);
+    if (isInside(mousePosition, rect)) {
+      redrawButton(ctx);
+      if (props.onClick && typeof props.onClick === 'function') {
+        props.onClick(event);
+      }
+    }
+  };
+  function trackMousePosition(canvas, event) {
```

Comment:
move this function outside of `renderButton`. A good place would be put into utils since can be reused by other elements later.